deploy.py

```diff
@@ -432,14 +436,25 @@ def create_tab_file(self, binaries):
     ]
     if self.args.verbose_build:
       elf2tab_args.append("--verbose")
+    stack_sizes = set()
     for arch, app_file in binaries.items():
       dest_file = os.path.join(self.tab_folder, "{}.elf".format(arch))
       shutil.copyfile(app_file, dest_file)
       elf2tab_args.append(dest_file)
+      # extract required stack size directly from binary
+      nm = self.checked_command_output(
+          ["nm", "--print-size", "--radix=x", app_file])
+      for line in nm.splitlines():
+        if "STACK_MEMORY" in line:
+          required_stack_size = int(line.split(" ", maxsplit=2)[1], 16)
+          if stack_sizes and required_stack_size not in stack_sizes:
```

Comment:
Why not simply adding to the set and, outside of the for loop, issue an error if `len(stack_sizes) != 1`?

Reply:
Done